### ISSUE TITLE
Any solid way to emulate surge's 200.html functionality locally?

### ISSUE THREAD

**Issue by howardroark** [sintaxi/surge#213]
Hey!

I was wondering if you had a standard way to emulate the 200.html functionality locally?  Perhaps something akin to Harp.

I think it's such a great idea, but it is a little awkward to upload each time you want to test.

Thanks!!!

**Comment by danreeves** [comment]
I wanted to do that recently and I ended up pulling the relevant middleware out of harp for easy use with express or browser-sync. Package and usage here: https://www.npmjs.com/package/lt-surge

**Comment by sintaxi** [comment]
@danreeves very cool. I was going to suggest `harp` but this looks better.

**Comment by danreeves** [comment]
Yeah, I was using harp for a while but wanted something lighter for when I have something else handling all the templating and sass

**Comment by howardroark** [comment]
I ended up using harp :) ... https://github.com/howardroark/dobutton

Didn't know it did that with `200.html`!  Been looking for a thing like this for a long time.

@danreeves  I'll have a look!  Though npm scripts make it pretty simple for combining the two ;)

**Comment by danreeves** [comment]
I used harp to do this for a long time but in the end wanted something without the weight of templating, asset compiling, and everything else harp does (things that can be a pain when you swap between node versions and take a long time to install). The middleware in that package is only the things that handle route resolution in regards to static assets.

I don't know if it covers surge 1 to 1, though.

**Comment by geelen** [comment]
@danreeves the only thing I can see is that if you have `/something.css` on Surge it 404s, whereas this responds with the fallback. Still nice though!

**Comment by geelen** [comment]
Oh also I can't figure out what harp does because I can't get it to install atm... Hence why it'd be nice to have a dedicated 'local-surge' project

**Comment by danreeves** [comment]
@geelen Thanks for letting me know; I've been meaning to tidy that module up and write some tests.
